riscv/src/register/marchid.rs

```diff
@@ -1,28 +1,9 @@
 //! marchid register
 
-use core::num::NonZeroUsize;
-
-/// marchid register
-#[derive(Clone, Copy, Debug)]
-pub struct Marchid {
-    bits: NonZeroUsize,
-}
-
-impl Marchid {
-    /// Returns the contents of the register as raw bits
-    #[inline]
-    pub fn bits(&self) -> usize {
-        self.bits.get()
-    }
+read_only_csr! {
```

Comment:
In case we want this new error variant, we can add a new macro/macro variant that, when trying to read/write, checks if the CSR is implemented

Reply:
Right, I went with the simpler implementation to open the conversation. I think encoding the semantics into the type may be better, as well.

To start, I think I'll just add the `Unimplemented` error variant, and the corresponding macro branch to the read-only helpers. The only registers using the `NonZeroUsize`/`Unimplemented` are all read-only.